### PR TITLE
feat : add Popover 

### PR DESCRIPTION
**Pull request by IliaSolovev**

### CODE REVIEW

File: src/components/forms/inputs/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -21,61 +21,50 @@ interface Props {
...
+              {isOpen && (
+                <ScrollContainer
+                  ref={ref}
+                  style={{
+                    top: intersectionSide === 'bottom' ? `-${dropdownHeight + listMargin}px` : `${dropdownLabelHeight + listMargin}px`,
+                  }}
+                >
+                  {items.map(({ label, value: itemValue }) => (
+                    <div
+                      tw="flex items-center pl-2 pr-8 text-monochrome-black text-14 leading-32 hover:bg-monochrome-light-tint"
```
Comment (RomanAchilov) [inline src/components/forms/inputs/dropdown/dropdown.tsx:52]: different paddings from design?

File: src/components/menu/menu.tsx [hunk 2]
```diff
@@ -20,6 +18,63 @@ interface Props {
...
+  bordered,
+  testContext = '',
+}: Props) => (
+  <Popover tw="text-monochrome-black">
+    {({ isOpen, setIsOpen }) => {
+      const { ref, intersectionSide } = useIntersectionSide({ dependency: [isOpen] });
+      const position = intersectionSide === 'bottom' ? 'top' : 'bottom';
+
+      return (
+        <>
```
Comment (RomanAchilov) [inline src/components/menu/menu.tsx:32]: useless React.Fragment

File: src/components/menu/menu.tsx [hunk 3]
```diff
@@ -20,6 +18,63 @@ interface Props {
...
+          <MenuIcon
+            onClick={() => setIsOpen(!isOpen)}
+            data-test={`menu:icon:${testContext}`}
+          >
+            {bordered ? <Icons.MoreOptionsWithBorder /> : <Icons.MoreOptions />}
+            {isOpen && (
+              <div
+                ref={ref}
+                style={{
```
Comment (RomanAchilov) [inline src/components/menu/menu.tsx:41]: it doesn't look good

File: src/components/menu/menu.tsx [hunk 4]
```diff
@@ -1,11 +1,9 @@
...
 import tw, { styled, css } from 'twin.macro';
 
 import { spacesToDashes } from '../../utils';
 import { Icons } from '../icon';
-import { useClickOutside } from '../../hooks';
+import { useIntersectionSide } from '../../hooks';
+import { Popover } from '../popover';
 
 export interface MenuItemType {
   label: string;
```
Comment (RomanAchilov) [inline src/components/menu/menu.tsx:9]: we have a mandatory Label, although we can pass Content

File: src/components/popover/popover.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  setIsOpen: Dispatch<boolean>;
+  isOpen: boolean;
+}
+
+export interface Props {
+  children: React.FC<ChildrenProps>;
+  className?: string;
+}
+
+export const Popover = ({
```
Comment (RomanAchilov) [inline src/components/popover/popover.tsx:15]: 👍

File: src/components/table/pagination.tsx [hunk 6]
```diff
@@ -131,59 +131,57 @@ export const Pagination = ({
...
-        </PaginationElements.Dots>
-        {popupIsOpen && (
-          <div tw="absolute" style={{ top: '-74px', left: '-46px' }}>
-            <Tooltip />
-          </div>
-        )}
-      </div>
-    );
-  };
+  const Ellipsis = () => (
```
Comment (RomanAchilov) [inline src/components/table/pagination.tsx:134]: lit looks great

File: src/components/tooltip/tooltip.tsx [hunk 7]
```diff
@@ -29,23 +31,16 @@ export const Tooltip = ({
...
-
-    return () => {
-      observer.disconnect();
-    };
   }, [isVisible]);
 
+  useIntersectionCallback({
+    callback: ([entry]) => {
+      !entry.isIntersecting &&
+      setTooltipPositionType(entry.intersectionRect.left ? 'top-left' : 'top-right');
```
Comment (RomanAchilov) [inline src/components/tooltip/tooltip.tsx:39]: what if you need an intersection from below or from above?

File: src/components/tooltip/tooltip.tsx [hunk 8]
```diff
@@ -95,35 +88,35 @@ const Message = styled.div<{type?: 'top-center' | 'top-right' | 'top-left' | 'le
     content: '';
     ${({ type }) => [
     type === 'top-left' && css`
-      bottom: -8px;
-      left: calc(100% - 8px);
-      border-right-color: #1b191b;
-      border-left: none;
-    `,
+        bottom: -8px;
```
Comment (RomanAchilov) [inline src/components/tooltip/tooltip.tsx:91]: Easy refactoring with TW jit

File: src/hooks/use-intersection-side.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
+import {
+  DependencyList, useLayoutEffect, useRef, useState,
+} from 'react';
+
+type Callback = (entries: IntersectionObserverEntry[]) => void;
+
+export interface Props {
```
Comment (RomanAchilov) [inline src/hooks/use-intersection-side.tsx:7]: u didn't use it

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/components/menu/menu.tsx (hunk 4 above)
Comment (IliaSolovev) [reply src/components/menu/menu.tsx:9]: Content it is wrapper over the label

**Pull request merged**